pkg/operator/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -35,45 +32,38 @@ func NewConfigObserver(
 	operatorConfigInformers operatorv1informers.SharedInformerFactory,
 	configInformers configinformers.SharedInformerFactory,
 	eventRecorder events.Recorder,
-) *ConfigObserver {
-	c := &ConfigObserver{
-		ConfigObserver: configobserver.NewConfigObserver(
-			operatorClient,
-			eventRecorder,
-			configobservation.Listers{
-				ResourceSync:        resourceSyncer,
-				APIServerLister_:    configInformers.Config().V1().APIServers().Lister(),
-				ImageConfigLister:   configInformers.Config().V1().Images().Lister(),
-				ProjectConfigLister: configInformers.Config().V1().Projects().Lister(),
-				ProxyLister_:        configInformers.Config().V1().Proxies().Lister(),
-				IngressConfigLister: configInformers.Config().V1().Ingresses().Lister(),
-				EndpointsLister:     kubeInformersForEtcdNamespace.Core().V1().Endpoints().Lister(),
-				SecretLister_:       kubeInformers.Core().V1().Secrets().Lister(),
-				PreRunCachesSynced: []cache.InformerSynced{
-					operatorConfigInformers.Operator().V1().OpenShiftAPIServers().Informer().HasSynced,
-					configInformers.Config().V1().Images().Informer().HasSynced,
-					configInformers.Config().V1().Projects().Informer().HasSynced,
-					configInformers.Config().V1().Proxies().Informer().HasSynced,
-					configInformers.Config().V1().Ingresses().Informer().HasSynced,
-				},
+) factory.Controller {
+	c := configobserver.NewConfigObserver(
+		operatorClient,
+		eventRecorder,
+		configobservation.Listers{
+			ResourceSync:        resourceSyncer,
+			APIServerLister_:    configInformers.Config().V1().APIServers().Lister(),
+			ImageConfigLister:   configInformers.Config().V1().Images().Lister(),
+			ProjectConfigLister: configInformers.Config().V1().Projects().Lister(),
+			ProxyLister_:        configInformers.Config().V1().Proxies().Lister(),
+			IngressConfigLister: configInformers.Config().V1().Ingresses().Lister(),
+			EndpointsLister:     kubeInformersForEtcdNamespace.Core().V1().Endpoints().Lister(),
+			SecretLister_:       kubeInformers.Core().V1().Secrets().Lister(),
+			PreRunCachesSynced: []cache.InformerSynced{
+				operatorConfigInformers.Operator().V1().OpenShiftAPIServers().Informer().HasSynced,
```

Comment:
I think that I would have to fix `configobserver.NewConfigObserver` so I left it for now.